security/pkg/nodeagent/sds/sdsservice.go

```diff
@@ -139,6 +155,38 @@ func (s *sdsservice) register(rpcs *grpc.Server) {
 	sds.RegisterSecretDiscoveryServiceServer(rpcs, s)
 }
 
+// DebugInfo serializes the current sds client data into JSON for the debug endpoint
+func (s *sdsservice) DebugInfo() (string, error) {
+	sdsClientsMutex.RLock()
+	defer sdsClientsMutex.RUnlock()
+
+	clientDebug := make([]sdsclientdebug, 0)
+	for connKey, conn := range sdsClients {
+		conn.mutex.RLock()
+		c := sdsclientdebug{
+			ConnectionID:     connKey.ConnectionID,
+			ProxyID:          conn.proxyID,
+			ResourceName:     conn.ResourceName,
+			CertificateChain: string(conn.secret.CertificateChain),
```

Comment:
conn.secret.CertificateChain and conn.secret.RootCert are byte arrays, can we parse them for better readability?

Perhaps distinguish an active connection and a stale connection, and a timestamp about the creation of an active connection, and a timestamp about when a connection becomes stale?